src/main/java/nl/tudelft/ewi/devhub/server/database/controllers/Commits.java

```diff
@@ -47,13 +57,17 @@ public Commits(final EntityManager entityManager, final RepositoriesApi reposito
 	public Commit ensureExists(RepositoryEntity repositoryEntity, String commitId) {
 		return retrieve(repositoryEntity, commitId).orElseGet(() -> {
 			final Commit commit = new Commit();
+			final DiffModel diffModel = retrieveDiffModel(repositoryEntity, commitId);
```

Comment:
Git server requests may fail. The question is how to handle it. Will we still create a commit, but with missing data? Or will we not create the commit at all? I am not sure about the best approach, but this approach is inconsistent with how we fetch the other commit information. See how `enhanceCommitSafely(...)` is invoked at line 68. Perhaps it is an idea to move this diff retrieval (and using its results in lines 65-66) to the same `enhanceCommitSafely` method. It will also fix your build (which fails because both `retrieveDiffModel` and `retrieveCommit` fail due to the lack of a git server mock.